### PR TITLE
add generic constructors for Acb and Arb (through Arf)

### PR DESCRIPTION
**Pull request by kalmarek**
Add general constructors for `Real`, which go through `Arf`.

this allows you to write `Arb(big(1.))` and `Acb(big(1.0), big(2))`

### CODE REVIEW

**Review by saschatimme — APPROVED** [review]

**Review by Joel-Dahne — APPROVED** [review]
I think it's very reasonable to have a general constructor for `Acb` which goes through `Arb`, this will always give rigorous results. However and `Arb` constructors which goes through `Arf` could potentially lead to the result not enclosing the input. It's fine as long as all constructors for `Arf` give exact results, at the moment this is the case so it should be fine. But in the future we have to remember to enforce this.

I can note that I have done a lot of this work in the clean-up of the constructors that I did. But if you want to get this done before the release we can merge this and then I'll take it into account when finishing the clean-up.

**Comment by kalmarek** [comment]
yeah, I'd like to have it in the released version, but this can also wait until your cleanup is merged (before we release, I suppose?)

**Comment by Joel-Dahne** [comment]
I don't think I'll finish the cleanup in time for the release. I'm perfectly fine just taking care of them later.

**Pull request merged**
